File: ops/scripts/upload-to-vector-store.py
#!/usr/bin/env python3

"""
Upload cancellation tickets to OpenAI Vector Store

This script converts our JSONL data to individual text files
and uploads them to the existing vector store.
"""

import json
import os
import time
from pathlib import Path
from openai import OpenAI

def upload_to_vector_store():
    print("🔄 Uploading cancellation tickets to vector store...")
    
    # Initialize OpenAI client
    client = OpenAI(api_key=os.getenv('OPENAI_API_KEY'))
    vector_store_id = os.getenv('OPENAI_VECTOR_STORE_ID')
    
    if not vector_store_id:
        raise ValueError("OPENAI_VECTOR_STORE_ID environment variable not set")
    
    # Read the JSONL data
    input_file = Path("vector-store-data-2025-10-10.jsonl")
    if not input_file.exists():
        raise FileNotFoundError(f"Input file {input_file} not found")
    
    documents = []
    with open(input_file, 'r', encoding='utf-8') as f:
        for line in f:
            if line.strip():
                documents.append(json.loads(line))
    
    print(f"Found {len(documents)} documents to upload")
    
    success_count = 0
    error_count = 0
    
    # Process documents in batches
    batch_size = 5  # Smaller batches to avoid rate limits
    for i in range(0, len(documents), batch_size):
        batch = documents[i:i + batch_size]
        batch_num = (i // batch_size) + 1
        total_batches = (len(documents) + batch_size - 1) // batch_size
        
        print(f"\n📦 Processing batch {batch_num}/{total_batches} ({len(batch)} documents)")
        
        for j, doc in enumerate(batch):
            global_index = i + j + 1
            try:
                # Create text content for the file
                metadata = doc.get('metadata', {})
                text_content = f"""CANCELLATION TICKET: {metadata.get('ticketId', 'unknown')}
Subject: {metadata.get('subject', 'N/A')}
Language: {metadata.get('language', 'unknown')}
Category: {'Cancellation' if metadata.get('is_cancellation') else 'Other'}
Payment Issue: {'Yes' if metadata.get('has_payment_issue') else 'No'}
Edge Case: {metadata.get('edge_case', 'none')}
Customer Concerns: {', '.join(metadata.get('customer_concerns', [])) or 'None'}

CONVERSATION:
{doc.get('text', '')}

METADATA:
- Ticket ID: {metadata.get('ticketId', 'unknown')}
- Language: {metadata.get('language', 'unknown')}
- Has Payment Issue: {metadata.get('has_payment_issue', False)}
- Edge Case: {metadata.get('edge_case', 'none')}
- Customer Concerns: {', '.join(metadata.get('customer_concerns', [])) or 'None'}
- Topic Keywords: {', '.join(metadata.get('topic_keywords', [])) or 'None'}
- Conversation Summary: {metadata.get('conversation_summary', 'None')}
"""
                
                # Create temporary file
                temp_filename = f"temp_cancellation_{metadata.get('ticketId', global_index)}.txt"
                with open(temp_filename, 'w', encoding='utf-8') as temp_file:
                    temp_file.write(text_content)
                
                # Upload file to OpenAI
                with open(temp_filename, 'rb') as temp_file:
                    file = client.files.create(
                        file=temp_file,
                        purpose='assistants'
                    )
                
                # Attach to vector store
                vector_store_file = client.vector_stores.files.create(
                    vector_store_id=vector_store_id,
                    file_id=file.id
                )
                
                # Clean up temp file
                os.remove(temp_filename)
                
                print(f"  ✅ {global_index}/{len(documents)}: {metadata.get('ticketId', 'unknown')} ({file.id})")
                success_count += 1
                
            except Exception as e:
                print(f"  ❌ {global_index}/{len(documents)}: Error uploading {metadata.get('ticketId', 'unknown')}: {str(e)}")
                error_count += 1
                
                # Clean up temp file if it exists
                temp_filename = f"temp_cancellation_{metadata.get('ticketId', global_index)}.txt"
                if os.path.exists(temp_filename):
                    os.remove(temp_filename)
        
        # Small delay between batches
        if i + batch_size < len(documents):
            print("⏳ Waiting 3 seconds before next batch...")
            time.sleep(3)
    
    print(f"\n🎉 Upload completed!")
    print(f"\n📊 Results:")
    print(f"   Total documents: {len(documents)}")
    print(f"   Successfully uploaded: {success_count}")
    print(f"   Errors: {error_count}")
    print(f"   Success rate: {(success_count / len(documents) * 100):.1f}%")
    
    print(f"\n🔗 Vector Store ID: {vector_store_id}")
    print(f"📁 Files uploaded with purpose: assistants")
    print(f"\n✅ Your RAG system is now ready to use the enhanced cancellation data!")

if __name__ == "__main__":
    try:
        upload_to_vector_store()
    except Exception as e:
        print(f"\n❌ Upload failed: {str(e)}")
        exit(1)

File: ops/scripts/upload-to-vector-store.js
#!/usr/bin/env tsx
"use strict";
/**
 * Upload cancellation tickets to OpenAI Vector Store
 *
 * This script converts our JSONL data to individual text files
 * and uploads them to the existing vector store.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
require("dotenv/config");
const fs_1 = require("fs");
const path_1 = require("path");
const openai_1 = __importDefault(require("openai"));
const inputFilePath = (0, path_1.join)(process.cwd(), 'vector-store-data-2025-10-10.jsonl');
const tempDir = (0, path_1.join)(process.cwd(), 'temp-vector-files');
async function uploadToVectorStore() {
    console.log(`🔄 Uploading cancellation tickets to vector store...`);
    try {
        // Read the JSONL data
        const jsonlContent = (0, fs_1.readFileSync)(inputFilePath, 'utf-8');
        const documents = jsonlContent
            .trim()
            .split('\n')
            .map(line => JSON.parse(line));
        console.log(`Found ${documents.length} documents to upload`);
        // Initialize OpenAI client
        const openai = new openai_1.default({ apiKey: process.env.OPENAI_API_KEY });
        const vectorStoreId = process.env.OPENAI_VECTOR_STORE_ID;
        if (!vectorStoreId) {
            throw new Error('OPENAI_VECTOR_STORE_ID environment variable not set');
        }
        // Clean up temp directory
        if (require('fs').existsSync(tempDir)) {
            (0, fs_1.rmSync)(tempDir, { recursive: true });
        }
        (0, fs_1.mkdirSync)(tempDir, { recursive: true });
        const uploadResults = [];
        let successCount = 0;
        let errorCount = 0;
        // Process documents in batches to avoid rate limits
        const batchSize = 10;
        for (let i = 0; i < documents.length; i += batchSize) {
            const batch = documents.slice(i, i + batchSize);
            console.log(`\n📦 Processing batch ${Math.floor(i / batchSize) + 1}/${Math.ceil(documents.length / batchSize)} (${batch.length} documents)`);
            const batchPromises = batch.map(async (doc, index) => {
                try {
                    const globalIndex = i + index;
                    const filename = `cancellation-ticket-${doc.metadata.ticketId}.txt`;
                    const filePath = (0, path_1.join)(tempDir, filename);
                    // Create text content for the file
                    const textContent = `CANCELLATION TICKET: ${doc.metadata.ticketId}
Subject: ${doc.metadata.subject || 'N/A'}
Language: ${doc.metadata.language || 'unknown'}
Category: ${doc.metadata.is_cancellation ? 'Cancellation' : 'Other'}
Payment Issue: ${doc.metadata.has_payment_issue ? 'Yes' : 'No'}
Edge Case: ${doc.metadata.edge_case || 'none'}
Customer Concerns: ${(doc.metadata.customer_concerns || []).join(', ') || 'None'}

CONVERSATION:
${doc.text}

METADATA:
- Ticket ID: ${doc.metadata.ticketId}
- Language: ${doc.metadata.language || 'unknown'}
- Has Payment Issue: ${doc.metadata.has_payment_issue || false}
- Edge Case: ${doc.metadata.edge_case || 'none'}
- Customer Concerns: ${(doc.metadata.customer_concerns || []).join(', ') || 'None'}
- Topic Keywords: ${(doc.metadata.topic_keywords || []).join(', ') || 'None'}
- Conversation Summary: ${doc.metadata.conversation_summary || 'None'}
`;
                    // Write temporary file
                    (0, fs_1.writeFileSync)(filePath, textContent, 'utf-8');
                    // Upload file to OpenAI
                    const file = await openai.files.create({
                        file: require('fs').createReadStream(filePath),
                        purpose: 'assistants'
                    });
                    // Attach to vector store
                    const vectorStoreFile = await openai.vector_stores.files.create({
                        vector_store_id: vectorStoreId,
                        file_id: file.id
                    });
                    // Clean up temp file
                    require('fs').unlinkSync(filePath);
                    console.log(`  ✅ ${globalIndex + 1}/${documents.length}: ${filename} (${file.id})`);
                    return { success: true, fileId: file.id, vectorStoreFileId: vectorStoreFile.id, ticketId: doc.metadata.ticketId };
                }
                catch (error) {
                    console.error(`  ❌ ${i + index + 1}/${documents.length}: Error uploading ${doc.metadata.ticketId}: ${error.message}`);
                    return { success: false, error: error.message, ticketId: doc.metadata.ticketId };
                }
            });
            // Wait for batch to complete
            const batchResults = await Promise.all(batchPromises);
            uploadResults.push(...batchResults);
            // Count results
            batchResults.forEach(result => {
                if (result.success) {
                    successCount++;
                }
                else {
                    errorCount++;
                }
            });
            // Small delay between batches to avoid rate limits
            if (i + batchSize < documents.length) {
                console.log(`⏳ Waiting 2 seconds before next batch...`);
                await new Promise(resolve => setTimeout(resolve, 2000));
            }
        }
        // Clean up temp directory
        (0, fs_1.rmSync)(tempDir, { recursive: true });
        console.log(`\n🎉 Upload completed!`);
        console.log(`\n📊 Results:`);
        console.log(`   Total documents: ${documents.length}`);
        console.log(`   Successfully uploaded: ${successCount}`);
        console.log(`   Errors: ${errorCount}`);
        console.log(`   Success rate: ${((successCount / documents.length) * 100).toFixed(1)}%`);
        if (errorCount > 0) {
            console.log(`\n❌ Errors encountered:`);
            uploadResults
                .filter(r => !r.success)
                .forEach(r => console.log(`   - ${r.ticketId}: ${r.error}`));
        }
        console.log(`\n🔗 Vector Store ID: ${vectorStoreId}`);
        console.log(`📁 Files uploaded with purpose: assistants`);
        console.log(`\n✅ Your RAG system is now ready to use the enhanced cancellation data!`);
    }
    catch (error) {
        console.error(`\n❌ Error uploading to vector store: ${error.message}`);
        process.exit(1);
    }
}
// Run the upload
uploadToVectorStore().catch(error => {
    console.error(`\n❌ Upload failed: ${error.message}`);
    process.exit(1);
});
